adam_preprocessing/shape_stroke_extraction.py

```diff
@@ -504,21 +553,63 @@ def get_strokes(self):
                         ).tolist(),
                         concept_name=self.obj_type,
                         confidence_score=1.0,
+                        # FIXME: I'm storing this for calculating touching;
+                        #  should do this in a more elegant way
+                        original_reduced_strokes=reduced_obj.tolist(),
                     ),
                     color=color.tolist(),
                     texture=None,
                     sub_part=None,
                 )
             )
-        self.data = data
+
+        # Calculate relative sizes using already-calculated absolute sizes:
+        for i in range(len(objects)):
+            relative_size = {'x_bigger_than': [], 'y_bigger_than': []}
+            for j in range(len(objects)):
+                if i == j:
+                    continue
+                else:
+                    other_object_name = objects[j]['object_name']
+                    for axis in ('x', 'y'):
+                        # Relative size is a discrete ('bigger_than') relation
+                        if objects[i]['size'][axis] > \
+                                objects[j]['size'][axis]:
+                            relative_size[f'{axis}_bigger_than'].append(other_object_name)
```

Comment:
It would be slightly simpler for the list here, `relative_size[f'{axis}_bigger_than']`, to be a list of booleans. Then you don't need the `if`.

Reply:
@spigo900 do you mean something like this:

```yaml
  relative_size:
    x_bigger_than:
    - false
    y_bigger_than:
    - true
```

(generated using commit 6ccd5b4)

Reply:
Yes, if the scene has two objects.